avocado/utils/cpu.py

```diff
@@ -188,3 +189,56 @@ def offline(cpu):
     if _get_cpu_status(cpu):
         return 1
     return 0
+
+
+def get_cpu_idle():
+    """
+    Get current cpu idle values
+    """
+    cpus_list = cpu_online_list()
+    states = range(len(glob.glob("/sys/devices/system/cpu/cpu0/cpuidle/state*")))
+    cpu_idle_state = [[0]*len(states)]*len(cpus_list)
+    for c in cpus_list:
+        for state in states:
+            state_file = "/sys/devices/system/cpu/cpu%s/cpuidle/state%s/disable" % (c, state)
+            try:
+                cpu_idle_state[c][state] = int(open(state_file).read())
+            except Exception:
+                pass  # Ignore if states are not available
+    return cpu_idle_state
+
+
+def set_cpu_idle(state="all", disable=True, old_value=None):
+    """
+    Set/Reset cpu idle states for all cpus
+
+    :param state: cpuidle state number, default: all states
+    :param disable: whether to disable/enable given cpu idle state
+    :param old_value: whether to reset to old value if given
```

Comment:
Well supplying custom matrix of values is actually quite useful not just for resetting the old value. How about tweaking the variable names to:

```
state_number: cpuidle state number, default: "all"
value: to be set value of "disable" entry of the given "state_number"
state: set state according to matrix using format provided by :mod:`get_cpuidle_states`
```

To me something like that would be easier to follow, but I'm not native, nor am I cpuidle expert. So it's just a recommendation to consider.

Reply:
also the docstring should mention that when `state` is given, `state_number` and `value` are ignored.